### PR TITLE
okta-changes-routes

### PR DESCRIPTION
**Pull request by yinstardev**
# Description

Okta auth Integration Changes

Fixes # (issue)

## Type of change

- [ x ] Routes Secure with JWT token from Backend. 
- [ x ] Token Handler Function to handle token once redirected from backend.
- [ x ] Get Full Access Token and User from Backend. 
- [ x ] Node Version Change. 
- [ x ] Readme File Update

( After node version change ) we have to do 

`export NODE_OPTIONS=--openssl-legacy-provider` To Run with new version. 

# How Has This Been Tested?

Deployed on Vercel and These things are working. Tested Manually.

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/api/getUserAndToken.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+const be_url = process.env.REACT_APP_BE_URL || '';
+
+const getJwtTokenFromLocalStorage = () => {
+  const token = localStorage.getItem('token');
+  return token;
+};
+
+export const fetchUserAndToken = async () => {
+  try {
```
Comment (sastaachar) [inline src/api/getUserAndToken.ts:22]: Dont use this , 
use the one i created in utils , no need to cache token now will do that later

File: src/components/router/RequireAuth.tsx [hunk 2]
```diff
@@ -1,12 +1,49 @@
...
-import { Navigate } from 'react-router-dom';
-import { useAppSelector } from '@app/hooks/reduxHooks';
-import { WithChildrenProps } from '@app/types/generalTypes';
+import React, { useEffect, useState, ReactElement } from 'react';
+import { Navigate, useLocation, Route, RouteProps } from 'react-router-dom';
+import axios from 'axios';
 
-const RequireAuth: React.FC<WithChildrenProps> = ({ children }) => {
-  const token = useAppSelector((state) => state.auth.token);
+const be_url = process.env.REACT_APP_BE_URL || '';
```
Comment (sastaachar) [inline src/components/router/RequireAuth.tsx:5]: move to environment.ts and give proper name

File: src/components/tables/BasicTable/BasicTable.tsx [hunk 3]
```diff
@@ -106,7 +106,12 @@ export const BasicTable: React.FC = () => {
           ],
         },
       ],
-      onFilter: (value: string | number | boolean, record: BasicTableRow) => record.name.includes(value.toString()),
+      onFilter: (value: boolean | React.Key, record: BasicTableRow) => {
+        if (typeof value === 'string') {
+          return record.name.includes(value);
```
Comment (sastaachar) [inline src/components/tables/BasicTable/BasicTable.tsx:111]: why is this used here ?

Comment (yinstardev) [reply src/components/tables/BasicTable/BasicTable.tsx:111]: It was giving typescript error.

**Pull request merged**